### ISSUE TITLE
Bug: Checkboxes should have a "disabled" visual appearance when `aria-disabled`

### ISSUE THREAD

**Issue by scurker** [dequelabs/cauldron#1752]
### Steps to reproduce

Have a checkbox with `aria-disabled`:

```tsx
<Checkbox aria-disabled={true} />
```

### Expected Behavior

The checkbox should appear as visually disabled. This can be important in scenarios where a tooltip may be necessary to help indicate why the checkbox is disabled and still allow access to the control for keyboard users.

### Version

v6.x

### Relevant log output

```shell

```
